src-electron/db/query-endpoint.js

```diff
@@ -335,6 +341,32 @@ async function deleteEndpoint(db, id) {
   return dbApi.dbRemove(db, 'DELETE FROM ENDPOINT WHERE ENDPOINT_ID = ?', [id])
 }
 
+async function getParentEndpointRef(db, parentEndpointIdentifier, sessionId) {
```

Comment:
I have a feeling that both of these functions could just be one function.
`getParentEndpointRefAndIdentifier` where you retrieve both with a single query. 

Less maintenance overall.

Reply:
Also add comments to your function. Is good practice.

Reply:
yeah, will do, thanks for catching that